src/origin_sdk/gql/queries/input_queries/session.py

```diff
@@ -0,0 +1,85 @@
+from typing import Dict
+from origin_sdk.gql.queries.input_queries.utils import RecursiveTree, tree_to_string
+
+
+def create_get_session_gql(variables: Dict[str, str], select: RecursiveTree):
+    """Creates a getSession query, since this is the most common wrapper for
+    queries.
+
+    Arguments:
+        variables: A dictionary of variables to be declared in the query
+        select: The projection/query body to specify.
+    """
+    
+    query: RecursiveTree = {
+        "getSession (sessionId: $sessionId)": select,
+    }
+    
+    return f'''query ( $sessionId: String!, {", ".join([f"""${
+        key
+    }: {
+        value
+    }""" for key, value in variables.items()])}) {
+        tree_to_string(query)
+    }'''
+
+
+config_tech_params_subtree = {
```

Comment:
Is there any way we can query the GQL schema to populate this set of fields? I feel like this is going to be another thing to maintain

Reply:
Yeah, I agree with the concern. I reckon we could roll forward with this and see what the overhead is to maintain this, though.

The only fields we currently _need_ here are `name`, `appliesToEndoBui` and `appliesToExo`. The queried field is an array, and all configured parameters will come back. So there is a small degree of maintenance that automatically happens. All the other fields are not important for the SDK at the moment.

The other fields are support for UI functionality that I'm not (as of now) trying to reimplement.